kyleshepherd/svelteui
# [@svelteui/core]: add IconRenderer

Just a test MR to get some feedback on the IconRenderer addition.

This MR attempts to fix the issue of the `icon` prop that is used across many SvelteUI components accepting a variable of type `HTMLOrSVGElement` but this causing an error due to no type-checking.

To test:
- Clone repo (make sure you have a `NPM_TOKEN` env var set
- In root of mono repo:
- `npm install --legacy-peer-deps`
- `npm run dev`
- Go to http://localhost:6006/?path=/story/components-iconrenderer--component

This story shows an Alert rendered with a Svelte component icon prop and a SVG Element icon prop
